state_machine/applications/flight/Tasks/radio.py

```diff
@@ -11,11 +11,18 @@
 
 ANTENNA_ATTACHED = False
 
+TX_SKIP = 5  # skip every TX_SKIP transmissions
+tx_ready_counter = 0
+
 def should_transmit():
     """
     Return if we should transmit
     """
-    return ANTENNA_ATTACHED and not tq.empty()
+    tx_ready = ANTENNA_ATTACHED and not tq.empty()
```

Comment:
I think we add one more case to this: 

```
tx_ready = ANTENNA_ATTACHED and not tq.empty() and cubesat.radio.fifo_empty()
```

Reply:
This should prevent us from starting a transmission while something is being received.

Reply:
I'm not against this, but since only one instance of the radio task is ever running, I'm not sure how that could ever happen anyway.

Reply:
I'm actually a little worried the FIFO might get stuck full or something

Reply:
The radio task runs every 0.5 seconds. So the maximum rate at which packets can be sent is every 0.5 seconds. After a packet has been transmitted the radio is left in listen mode. It can receive data in this mode, and it will hold onto that data in the FIFO until the receive function is called and it is read off the radio. Whenever the radio task takes less than 0.5 seconds to complete, this is a possibility. Given the delays to wait for acknowledges, this may not actually happen when the tx queue has stuff in it, but it is still a possibility that something can be added to the tx queue while something is being received. This would prevent that.

Reply:
Makes sense.